### ISSUE TITLE
Windows 10, cygwin, node 10.15.3 make-js install: ERR! OMG Cannot read property 'verbose' of undefined

### ISSUE THREAD

**Issue by Dzenly** [cmake-js/cmake-js#167]
From cygwin bash:
```
$ cmake-js install
[ 'C:\\Program Files\\nodejs\\node.exe',
  'C:\\Program Files\\nodejs\\node_modules\\cmake-js\\bin\\cmake-js',
  'install' ]
ERR! OMG Cannot read property 'verbose' of undefined

$ node --version
v10.15.3
```

**Comment by unbornchikken** [comment]
What is the output of:

```bash
cmake-js install --log-level=verbose
```

**Comment by jalfd** [comment]
It looks like there are several issues. The immediate problem is in CMake.getGenerators, where `this.log` is undefined, so instead of logging the message `"TOOL", "Failed to query CMake capabilities (CMake is probably older than 3.7)"`, it just fails instead.

The second problem in my case at least, is that I specify the -c parameter to use a custom CMake path, and that path has a space in it (C:\Program Files\cmake\...), and apparently cmake-js does not quote the supplied path before executing it, so it ends up trying to invoke `C:\Program -E capabilities`.

**Comment by unbornchikken** [comment]
Good catch. Would you help me to fix it? A PR would be very much appreciated.

**Comment by jalfd** [comment]
I'm afraid that I'm not very familiar with the cmake-js code base and it would take me more time to figure out how to fix this properly than I'm currently able to devote to it. I just ran into this problem and spent 10 minutes trying to diagnose what goes wrong, but actually _fixing_ it in a good way is not something I'd be able to do right now.

**Comment by jalfd** [comment]
> Good catch. Would you help me to fix it? A PR would be very much appreciated.

I've made a couple of PRs to fix this (#226, #227)

**Comment by Julusian** [comment]
Fixed in v6.2.0 by https://github.com/cmake-js/cmake-js/commit/3ee62bb16d37bcf4c128ae0bdedf322a63e3cb3a
